app/Insightly/Listeners/SyncContact.php

```diff
@@ -0,0 +1,127 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Insightly\Listeners;
+
+use App\Domain\Contacts\Contact;
+use App\Domain\Contacts\Events\ContactCreated;
+use App\Domain\Contacts\Events\ContactUpdated;
+use App\Domain\Contacts\Repositories\ContactRepository;
+use App\Insightly\Exceptions\ContactCannotBeUnlinked;
+use App\Insightly\InsightlyClient;
+use App\Insightly\InsightlyMapping;
+use App\Insightly\Repositories\InsightlyMappingRepository;
+use App\Insightly\Resources\ResourceType;
+use App\Insightly\SyncIsAllowed;
+use Illuminate\Bus\Queueable;
+use Illuminate\Contracts\Queue\ShouldQueue;
+use Illuminate\Support\Arr;
+use Psr\Log\LoggerInterface;
+
+final class SyncContact implements ShouldQueue
```

Comment:
Should we also implement the `failed` method to get a nice processable error when it fails?